pkg/testing/tools/artifacts_api.go

```diff
@@ -178,6 +181,27 @@ func (aac ArtifactAPIClient) GetBuildsForVersion(ctx context.Context, version st
 	return checkResponseAndUnmarshal[VersionBuilds](resp)
 }
 
+func (aac ArtifactAPIClient) FindBuild(ctx context.Context, version, excludeHash string, offset int) (buildDetails *BuildDetails, err error) {
```

Comment:
[Blocker]
Please, could you document what does `offset` do? It's not trivial to understand without reading the code and even reading the code it isn't clear why one would set it or hot to know which offset to use

Reply:
I'm trying to understand how one would know which offset to set if a priori the number of builds isn't known.

Reply:
Documented in https://github.com/elastic/elastic-agent/pull/4191/commits/d0dea5d98a4894dd004ebeeebb3e08ab3cb62fc2